extensions/github/src/links.ts

```diff
@@ -70,15 +70,15 @@ function getFileAndPosition(context: LinkContext): IFilePosition | INotebookPosi
 			const cellIndex = cell?.index ?? vscode.window.activeNotebookEditor.selection.start;
 
 			let range;
-			if (lineNumber !== undefined) {
+			if (lineNumber && (!vscode.window.activeTextEditor || vscode.window.activeTextEditor.selection.isEmpty || !vscode.window.activeTextEditor.selection.contains(new vscode.Position(lineNumber - 1, 0)))) {
 				range = new vscode.Range(new vscode.Position(lineNumber - 1, 0), new vscode.Position(lineNumber - 1, 1));
 			} else if (cell !== undefined) {
 				range = vscode.window.activeTextEditor?.selection;
 			}
 			return { type: LinkType.Notebook, uri, cellIndex, range };
 		} else {
 			// the active editor is a text editor
-			range = lineNumber !== undefined ? new vscode.Range(lineNumber - 1, 0, lineNumber - 1, 1) : vscode.window.activeTextEditor?.selection;
+			range = lineNumber !== undefined && (!vscode.window.activeTextEditor || vscode.window.activeTextEditor.selection.isEmpty || !vscode.window.activeTextEditor.selection.contains(new vscode.Position(lineNumber - 1, 0))) ? new vscode.Range(lineNumber - 1, 0, lineNumber - 1, 1) : vscode.window.activeTextEditor?.selection;
```

Comment:
Should this condition live in a function since it's quite long and used twice? (and I think it's already accidentally different)